scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -741,6 +742,48 @@ def small_sample_generator(n_dims):
                     hypotest(*samples, *args, axis=axis, **kwds)
 
 
+@pytest.mark.parametrize("axis", [0, 1])
+@pytest.mark.parametrize(("hypotest", "args", "kwds", "n_samples", "n_outputs",
+                          "paired", "unpacker"), axis_nan_policy_cases)
+def test_non_broadcastable(hypotest, args, kwds, n_samples, n_outputs, paired,
+                           unpacker, axis):
+    # test for correct error message when shapes are not broadcastable
+    if n_samples == 1:  # broadcasting only needed with >1 sample
+        return
+
+    # It's much easier to write a relatively comprehensive test here
+    # by relying on randomly-generated shapes to not be broadcastable.
+    # If we were to use the same seed all the time, the test would
+    # be no stronger than having chosen a single case, so change the
+    # seed every day. Any failing case is still reproducible.
+    seed = int(datetime.date.today().strftime("%Y%m%d"))
+    rng = np.random.default_rng(seed)
+    samples = [rng.random(size=rng.integers(2, 100, size=2))
+               for i in range(n_samples)]
```

Comment:
I agree that we can tune down the complexity of the test suite in time. Originally, these tests were written as they are because they helped me catch bugs. Perhaps they are not all needed once the decorator has reached a certain level of maturity. We certainly do need to start reducing the time it takes, probably by running only on certain functions.
Regarding this test, of course I didn't write the test to be complex for the sake of complexity but for completeness. Given the number of issue reports there have been over the years, clearly bugs relating to `axis`/`nan_policy` etc can be hard to detect with limited testing. So I'll let you make the call here. If you would like to simplify this, please go ahead and make the change, then ping me before merge. Thanks!

Reply:
Made a change to the test. let me know if it looks good to you.

> Regarding this test, of course I didn't write the test to be complex for the sake of complexity but for completeness. Given the number of issue reports there have been over the years, clearly bugs relating to `axis`/`nan_policy` etc can be hard to detect without extensive property-based testing.

I agree. Extensive testing is incredibly helpful in discovering new bugs and we should locally keep doing this. But now that there's a lot of solid tests confirming the decorator works, there's diminishing returns against stringent tests. Fixing the failing tests in the future could become a nightmare (for eg. porting the decorator to Array API) if everything is parametrized and run with randomly generated data! Hope this makes sense.

Reply:
We discussed offline and agreed to keep both but make the extensive test `xslow`. I'll make this change and figure out what's failing (besides lint).